flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSink.java

```diff
@@ -106,18 +109,29 @@ protected DynamoDbSink(
                 maxBatchSizeInBytes,
                 maxTimeInBufferMS,
                 maxRecordSizeInBytes);
-        checkNotNull(
-                tableName,
-                "Destination table name must be set when initializing the DynamoDB Sink.");
-        checkArgument(
-                !tableName.isEmpty(),
-                "Destination table name must be set when initializing the DynamoDB Sink.");
         checkArgument(
                 maxBatchSize <= 25,
                 "DynamoDB client supports only up to 25 elements in the batch.");
         checkNotNull(dynamoDbClientProperties, "DynamoDB client properties must be set.");
+
+        // Ensure that either tableName or tableArn is set. If both are set, tableArn takes
+        // precedence.
+        if (tableArn != null) {
+            final Arn arn = Arn.fromString(tableArn);
+            this.tableArn = arn.toString();
+            this.tableName = arn.resource().resource();
+        } else {
+            this.tableArn = null;
+            checkNotNull(
+                    tableName,
+                    "Destination table name must be set when initializing the DynamoDB Sink.");
```

Comment:
Should we update the error message to mention both ARN and name as possible options?